string filename
string directory
---
bool success